thorlcr/msort/tsorts1.cpp

```diff
@@ -336,23 +336,33 @@ protected: friend class CSortMerge;
         DBGLOG("CSortTransferServerThread started port %d",slave.getTransferPort());
         unsigned numretries = 10;
         try {
-            while (!term) {
+            while (!term)
+            {
                 ISocket* socket = server->accept(true);
-                if (!socket) {
+                if (!socket)
                     break;
-                }
 
                 rowcount_t poscount=0;
                 rowcount_t numrecs=0;
                 ISocketRowWriter *strm=NULL;
-                try {
+                try
+                {
                     waitRowIF();
                     strm = ConnectMergeWrite(rowif,socket,0x100000,poscount,numrecs);
                 }
-                catch (IJSOCK_Exception *e) { // retry if failed
-                    PrintExceptionLog(e,"WARNING: Exception(ConnectMergeWrite)");
+                catch (IJSOCK_Exception *e) // retry if failed
+                {
+                    PrintExceptionLog(e, "WARNING: Exception(ConnectMergeWrite)");
                     if (--numretries==0)
-                        throw e;
+                        throw;
+                    e->Release();
```

Comment:
Should the socket be closed here before continuing ?

Reply:
yes I think the old exception handling and the new, both leak the ISocket.
They should close and release it before continuing.